### DISCUSSION TITLE
Rotation

### DISCUSSION THREAD

**Discussion by Polygamoose** [Saibot393/Rideable#56] [Ideas]
Hello! I'm working on starting to prep for a WH40k Only War campaign where my PCs will be in control of a tank. Think along the lines of the movie "Fury" or "T-34". 

Currently, you have Rotation Lock which makes the rider unable to rotate unless the mount rotates. I was hoping that you could implement a feature that would cause riders to rotate along with their mount, but would also allow the rider to rotate on their own without the mount rotating. This would reflect a tank turret's ability to rotate independently of the tank's body, but the turret will also rotate when the tank body rotates.

Also, it would be really great if this could be a per-token setting instead of "all or none".

**Comment by Saibot393** [comment]
There is a feature called "riders that can move freely" which can be set on the ridden token. This feature can almost do what you described (rotate the rider with the ridden token but also allowing the rider to rotate independently). The problem would currently probably be, that that also means, that the rider/the turret could move freely within the ridden/tank. I could add a macro to lock the current position of a free riding token so that it can only rotate freely. Would that work for your purpose?

**Reply by Polygamoose** [reply]
It doesn't work exactly as you describe. With "riders than can move freely", the turret doesn't rotate when the tank base rotates, it always faces the same direction until I manually rotate it. Also, the tank has two sponson guns that, if I have "riders that can move freely" turned on, flop around the tank like a dying fish. If it's turned off, they stay where they belong. So having RTCMF turned on isn't the best solution for my problem.

That said, thank you so much for responding!! I'm very grateful.

**Reply by Saibot393** [reply]
Regarding the first part with the turret not rotating with the tank: Do you have "Link rotations" activated?
Ragarding th "flopping around" problem: Are you using modules that affect rotation such as auto rotate or about face and does this happen with "Link rotations" active?

**Reply by Saibot393** [reply]
I have added macro to lock the position of certain tokens for "Riders can move freely" with v2.18.0, which should in theory work for the case you described once that bug is solved

**Comment by demotech89** [comment]
Is there anyway to unmount the token just by moving it outside the tile after being mounted? Currently, it is moving the tile whenever I move the token. I am trying to create a car seat where players to mount in but get out of the vehicle without tapping "N"

Thanks for any help on this!
